### PR TITLE
Kd format multi record vc

### PR DESCRIPTION
**Pull request by karin-**
@wtachau

### CODE REVIEW

File: MultiRecordViewController.m [hunk 1]
```diff
@@ -41,32 +48,66 @@ - (UITableView*)recordTableView {
...
+}
+
 #pragma mark - Lifecycle
 
-- (instancetype)initWithRecords:(NSArray*)records {
+- (instancetype)initWithRecords:(EmailRecordGroup*)records {
   self = [super init];
   if (self) {
     self.records = records;
+
```
Comment (wtachau) [inline MultiRecordViewController.m]: :(

File: Medlink/EmailRecord.m [hunk 2]
```diff
@@ -70,4 +71,5 @@ - (instancetype)initWithSortType:(InboxSortType)sortType record:(EmailRecord*)re
   return self;
 }
 
+
```
Comment (wtachau) [inline Medlink/EmailRecord.m]: :(

File: MultiRecordViewController.m [hunk 3]
```diff
@@ -84,58 +129,112 @@ - (void)setup {
   if ([self respondsToSelector:@selector(edgesForExtendedLayout)])
     self.edgesForExtendedLayout = UIRectEdgeNone;
   
-  self.navigationItem.title = ((EmailRecord*)[self.records firstObject]).eventName;
+  NSDateFormatter* dateFormatter;
+  NSDate* date;
+  dateFormatter = [[NSDateFormatter alloc] init];
```
Comment (wtachau) [inline MultiRecordViewController.m]: any reason not to make this in one line?
`NSDateFormatter* dateFormatter = [[NSDateFormatter alloc] init];`?

Comment (wtachau) [inline MultiRecordViewController.m]: Also, we don't need the dateFormatter unless we are in the `InboxSortTypeDate` block, correct? We can just initialize it in that case statement then

File: MultiRecordViewController.m [hunk 4]
```diff
@@ -84,58 +129,112 @@ - (void)setup {
...
   [self.shareButton pinToTopWithPadding:recordTableViewPadding];
   [self.shareButton pinToRightWithPadding:recordTableViewPadding];
   
-  [self.recordTableView placeBelowView:self.dateRangeLabel padding:recordTableViewPadding];
+  UILabel* subHeader;
+  switch (self.records.sortType) {
+    case InboxSortTypeDate:
+      // what should go here?
+      subHeader = [[UILabel alloc] init];
+      [self.view addSubview:subHeader];
```
Comment (wtachau) [inline MultiRecordViewController.m]: did you figure out that question? :) can you help me understand the reasoning for this

File: MultiRecordViewController.m [hunk 5]
```diff
@@ -84,58 +129,112 @@ - (void)setup {
...
+      subHeader = self.dateRangeLabel;
+      break;
+    case InboxSortTypeDoctor:
+      subHeader = self.specialtyLabel;
+  }
+  [subHeader sizeToFit];
+  
+  if (self.records.sortType == InboxSortTypeDoctor) {
+    _doctorImageView.layer.borderWidth = 2;
+    _doctorImageView.layer.borderColor = [UIColor skyBlueColor].CGColor;
```
Comment (wtachau) [inline MultiRecordViewController.m]: any reason we're not calling `self.doctorImageView`?

Comment (wtachau) [inline MultiRecordViewController.m]: Also, unless these are dependent on something in this if/else, let's move those lines to the lazy getter. Save this section for the layout

**Comment by wtachau** [comment]
the result looks great. let's clean it up a little bit and then we can merge it.

File: MultiRecordViewController.m (hunk 3 above)
Comment (karin-) [inline MultiRecordViewController.m]: I wasn't able to declare a dateFormatter inside the case statement :(  not sure why bc I don't know obj c very well, but I'll look it up

File: MultiRecordViewController.m (hunk 4 above)
Comment (karin-) [inline MultiRecordViewController.m]: haha nope. I was wanting to put a subhead there, but I couldn't find a mock for records sorted by date and I also couldn't think of anything that made sense. I ended up assuming every sort type would have a subhead and just left that one blank.

the end result is definitely weird - should I take it out?

File: MultiRecordViewController.m (hunk 5 above)
Comment (karin-) [inline MultiRecordViewController.m]: Yeah I was wondering about how to set this up. If I do this in the lazy getter, then I think I'd have to add the doctorImageView to the subview here, in viewWillLayoutSubviews - is that an acceptable place to add a subview? Not clear on when that kind of thing should and shouldn't happen.

File: MultiRecordViewController.m (hunk 4 above)
Comment (wtachau) [inline MultiRecordViewController.m]: Yeah, I see your point. The fun of having ambiguous mocks :)

I would just get rid of the InboxSortTypeDate case.

File: MultiRecordViewController.m (hunk 3 above)
Comment (wtachau) [inline MultiRecordViewController.m]: Resolved. One of ObjC's fun bugs :)

File: MultiRecordViewController.m [hunk 6]
```diff
@@ -41,29 +48,64 @@ - (UITableView*)recordTableView {
...
+  _eventNameLabel.font = [UIFont fontWithName:NSLocalizedString(@"FONT_BOLD", nil) size:20];
+  _eventNameLabel.textColor = [UIColor darkBlueColor];
   return _eventNameLabel;
 }
 
 - (UILabel*)dateRangeLabel {
   if (_dateRangeLabel) return _dateRangeLabel;
   _dateRangeLabel = [[UILabel alloc] init];
-  _dateRangeLabel.font = [UIFont fontWithName:@"HelveticaNeue" size:14];
+  _dateRangeLabel.font = [UIFont fontWithName:NSLocalizedString(@"FONT", nil) size:14];
```
Comment (aarondaub) [inline MultiRecordViewController.m]: Localized strings shouldn't be used for font names

**Comment by wtachau** [comment]
:+1:

**Pull request merged**